src/y-mongodb.js

```diff
@@ -101,13 +101,27 @@ export class MongodbPersistence {
 		return this._transact(docName, async (db) => {
 			const updates = await U.getMongoUpdates(db, docName);
 			const ydoc = new Y.Doc();
+			let applyNum = 0;
 			ydoc.transact(() => {
 				for (let i = 0; i < updates.length; i++) {
-					Y.applyUpdate(ydoc, updates[i]);
+					try {
+						Y.applyUpdate(ydoc, updates[i]);
+					} catch (e) {
+						console.warn(`Failed to apply update ${i} to document "${docName}".`, e);
+					}
+					applyNum += 1;
 				}
 			});
-			if (updates.length > this.flushSize) {
+			if (updates.length > this.flushSize && applyNum === updates.length) {
+				// 情况1: 更新数量超过阈值并且全部应用成功，执行文档刷新
 				await U.flushDocument(db, docName, Y.encodeStateAsUpdate(ydoc), Y.encodeStateVector(ydoc));
+			} else if (applyNum === updates.length) {
+				/* empty */
+			} else if (applyNum !== updates.length) {
+				// 情况3: 无法应用所有更新，记录警告
+				console.warn(
+					`Failed to apply all updates to document "${docName}". Applied ${applyNum}/${updates.length} updates.`,
+				);
 			}
```

Comment:
These if/else blocks can be easily simplified:

```suggestion
			if (updates.length > this.flushSize && ) {
				if (applyNum === updates.length) {
					await U.flushDocument(db, docName, Y.encodeStateAsUpdate(ydoc), Y.encodeStateVector(ydoc));
				} else {
					console.warn(
					`Failed to apply all updates to document "${docName}". Applied ${applyNum}/${updates.length} updates.`,
				);
				}
			}
```